### ISSUE TITLE
Moran's I can produce out of range values

### ISSUE THREAD

**Issue by alexgenin** [spatial-ews/spatialwarnings#105]
```r
m <- cbind(matrix(1, nrow = 100, ncol = 50), 
           matrix(0, nrow = 100, ncol = 50))

raw_moran(m)
# returns 1.010208
```

Looking quickly at the [code](https://github.com/spatial-ews/spatialwarnings/blob/master/src/moran_correlation.cpp), this may be a normalization issue, where the mean/variance are computed on the whole matrix but the individual terms in the sum are computed only on an inner matrix obtained by discarding the last and first row/column.

**Comment by alexgenin** [comment]
Potential fix in 5ab681bb5f9d468d2ce319fc75c8351330e9853a, but it would be good to test against another implementation of moran correlation.

**Comment by alexgenin** [comment]
We now test our implementation of Moran's I against raster::Moran() (added in 4d73ef7fa4f88a293819ff9c75e2b66bec50c1a9).

**Comment by alexgenin** [comment]
This seems to be fixed now, and deployed in spw v3.0.2 (on its way to CRAN), so I'm closing this issue now.
